libs/api/domains/translations/src/lib/translations.service.ts

```diff
@@ -27,32 +41,74 @@ const errorHandler = (name: string) => {
 export class TranslationsService {
   constructor(private contentfulRepository: ContentfulRepository) {}
 
-  getTranslations = async (
-    namespaces?: string[],
-    lang?: string,
-  ): Promise<TranslationsDict | null> => {
-    const locale = locales.find((l) => l.code === localeMap[lang])
-
-    const result = await this.contentfulRepository
-      .getLocalizedEntries<any>('*', {
-        ['content_type']: 'namespace',
-        select: 'fields.strings',
-        'fields.namespace[in]': namespaces.join(','),
-      })
-      .catch(errorHandler('getNamespace'))
-
-    return result.items.reduce((acc, cur) => {
-      const strings = cur.fields.strings
-
-      return {
-        ...acc,
-        ...mergeWith(
-          {},
-          locale.fallbackCode ? strings?.[locale.fallbackCode] : {},
-          strings?.[locale.code],
-          (o, s) => (isEmpty(s) ? o : s),
-        ),
+  getNamespaceMessages = memoize(
+    async (namespace: string) => {
+      const results = await this.contentfulRepository
+        .getLocalizedEntries<NamespaceFields>('*', {
+          ['content_type']: 'namespace',
+          select: 'fields.strings',
+          'fields.namespace': namespace,
+        })
+        .catch(errorHandler('getNamespace'))
+
+      const messages = {
+        id: namespace,
+        is: {},
+        en: {},
+      } as Messages
+
+      for (const item of results.items) {
+        for (const contentfulLocale of Object.keys(item.fields.strings ?? {})) {
+          const strings = item.fields.strings
+
+          if (!strings) {
+            continue
+          }
+
+          for (const key of Object.keys(strings[contentfulLocale])) {
+            const locale = locales.find(
+              (item) => item.code === contentfulLocale,
+            )!.locale as Locale
```

Comment:
Can move the `locale` variable one level up, as well as caching the locale strings:

```
const localeStrings = strings[contentfulLocale] ?? {}
```

Can also cache `defaultStrings` one level higher.